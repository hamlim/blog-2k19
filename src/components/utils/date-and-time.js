/* eslint-disable */

export function monthNumberToMonthName(month) {
  switch (month) {
    case '01':
      return 'January'
    case '02':
      return 'February'
    case '03':
      return 'March'
    case '04':
      return 'April'
    case '05':
      return 'May'
    case '06':
      return 'June'
    case '07':
      return 'July'
    case '08':
      return 'August'
    case '09':
      return 'September'
    case '10':
      return 'October'
    case '11':
      return 'November'
    case '12': 
      return 'December'
  }
}

export function parseDate(dateString) {
  // MM-DD-YYYY
  let split = dateString.split('-');
  // DD-MM-YYYY
  split = [split[1], split[0], split[2]];
  split[1] = monthNumberToMonthName(split[1]);
  const attemptedString = split.join(' ');
  return new Date(attemptedString);
}