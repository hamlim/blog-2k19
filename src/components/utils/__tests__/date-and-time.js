import { parseDate, monthNumberToMonthName } from '../date-and-time.js'

test('parseDate returns a date object', () => {
  const mockDate = '07-20-1994'
  expect(parseDate(mockDate)).toMatchInlineSnapshot(`1994-07-20T04:00:00.000Z`)
})
