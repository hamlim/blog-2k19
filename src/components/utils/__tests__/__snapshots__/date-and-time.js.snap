// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`monthNumberToMonthName returns a readable month name 1`] = `"January"`;

exports[`monthNumberToMonthName returns a readable month name 2`] = `"February"`;

exports[`monthNumberToMonthName returns a readable month name 3`] = `"March"`;

exports[`monthNumberToMonthName returns a readable month name 4`] = `"April"`;

exports[`monthNumberToMonthName returns a readable month name 5`] = `"May"`;

exports[`monthNumberToMonthName returns a readable month name 6`] = `"June"`;

exports[`monthNumberToMonthName returns a readable month name 7`] = `"July"`;

exports[`monthNumberToMonthName returns a readable month name 8`] = `"August"`;

exports[`monthNumberToMonthName returns a readable month name 9`] = `"September"`;

exports[`monthNumberToMonthName returns a readable month name 10`] = `"October"`;

exports[`monthNumberToMonthName returns a readable month name 11`] = `"November"`;

exports[`monthNumberToMonthName returns a readable month name 12`] = `"December"`;
